gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/push/PushController.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.push;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.validation.Validated;
+import jakarta.validation.Valid;
+import reactor.core.publisher.Mono;
+
+/**
+ * A {@link Controller} implementation for handling PubSub Push JSON messages.
+ *
+ * <p>
+ * If push message handling is enabled, and the required HTTP dependencies are available on the classpath, this controller will
+ * handle all incoming push messages via a single URL path. The default path is {@code /push}. This is the path that should be
+ * configured in the GCP PubSub service.
+ * </p>
+ *
+ * <p>
+ * The incoming JSON messages contain metadata about the subscription from which they originated, and they will be routed to
+ * the corresponding {@link io.micronaut.gcp.pubsub.annotation.PushSubscription} method for the subscription.
+ * </p>
+ *
+ * @author Jeremy Grelle
+ * @since 5.4.0
+ */
+@Requires(beans = PubSubPushConfigurationProperties.class)
```

Comment:
why require a bean of type `PubSubPushConfigurationProperties`? That bean is not used in this controller.

Reply:
The intention is to disable this controller if the user sets `gcp.pubsub.push.enabled=false`. When that property is set to false, then the `PubSubPushConfigurationProperties` bean is disabled.

Is there a preferred way to achieve the same result?

Reply:
I think you should at least change it to `PushControllerConfiguration`

Reply:
[4dafe6d](https://github.com/micronaut-projects/micronaut-gcp/pull/1012/commits/4dafe6d2e6750b70ef669f1ac3413e97f83b6698)